rs/index/src/ivf/files/invalidated_ids.rs

```diff
@@ -0,0 +1,123 @@
+use std::fs::OpenOptions;
+
+use anyhow::{anyhow, Result};
+use memmap2::MmapMut;
+
+pub struct InvalidatedIdsStorage {
+    mmap: MmapMut,
+    num_points: usize,
+}
+
+const BITS_IN_BYTE: usize = 8;
+
+impl InvalidatedIdsStorage {
+    pub fn new(base_directory: &str, num_points: usize) -> Result<Self> {
+        let file = OpenOptions::new()
+            .read(true)
+            .write(true)
+            .create(true)
+            .open(format!("{}/invalidated_ids", base_directory))?;
+
+        let bytes_needed = num_points.div_ceil(BITS_IN_BYTE);
+        file.set_len(bytes_needed as u64)?;
+
+        let mmap = unsafe { MmapMut::map_mut(&file)? };
+        Ok(Self { mmap, num_points })
+    }
+
+    pub fn invalidate(&mut self, point_id: u32) -> Result<()> {
+        let point_id = point_id as usize;
+        if point_id >= self.num_points {
+            return Err(anyhow!("ID out of bound"));
+        }
+
+        let byte_index = point_id / BITS_IN_BYTE;
+        let bit_index = point_id % BITS_IN_BYTE;
+
+        self.mmap[byte_index] |= 1 << bit_index;
+
+        self.mmap.flush()?;
+        Ok(())
```

Comment:
you can probably skip the `Ok(())` if you remove the semicolon?

```suggestion
        self.mmap.flush()?
```